### PR TITLE
Improve safari development documentation

### PR DESCRIPTION
**Pull request by Hinton**
## Objective

Streamline the Safari development steps, putting more focus on the Xcode approach since it's by far nicer to work with.

### CODE REVIEW

File: docs/getting-started/clients/browser/index.md [hunk 1]
```diff
@@ -64,31 +64,57 @@ the Bitwarden heading in the Temporary Extensions page. To debug the popup:
...
+
+1. Build the extension:
+
+   ```bash
+   npm run build:watch
+   ```
+
+2. Edit `build/manifest.json`. Move the `nativeMessaging` permission from the `optional_permissions`
+   section into the `permissions` section
+3. Edit `build/index.html`, replace `<html class="__BROWSER__">` to `<html class="browser_safari">`.
```
Comment (MGibson1) [inline docs/getting-started/clients/browser/index.md:102]: This is fine for now, but we should automate this. https://bitwarden.atlassian.net/browse/TDL-215

File: docs/getting-started/clients/browser/index.md [hunk 2]
```diff
@@ -64,31 +64,57 @@ the Bitwarden heading in the Temporary Extensions page. To debug the popup:
...
+not automatically reload.
+
+:::
 
-To build and load the Browser extension:
+#### Production build
+
+The other alternative is to use the "proper" build process through gulp. This method doesn't require
+any manual processing of the output since gulp does it for us. However we have to completely rebuild
+the extension for every change which is slower.
```
Comment (MGibson1) [inline docs/getting-started/clients/browser/index.md:117]:
```suggestion
the extension for every change, which is slower.
```

**Review by MGibson1 — COMMENTED** [review]
very small item

File: docs/getting-started/clients/browser/index.md (hunk 1 above)
Comment (Hinton) [reply docs/getting-started/clients/browser/index.md:102]: I know, I started looking into it and then couldn't be bothered.

Comment (MGibson1) [reply docs/getting-started/clients/browser/index.md:102]: yup, that's why I made a ticket 👍

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**